src/Socket/messages-send.ts

```diff
@@ -883,14 +632,11 @@ export const makeMessagesSocket = (config: SocketConfig) => {
 				}
 
```

Comment:
The best way to solve the problem in the e2e test (regression) is to centrally resolve PN JIDs to their LID counterparts right before any session handling logic is executed. Check:

```suggestion
				// Resolve PN JIDs to LID JIDs before session handling
				await Promise.all(
					devices.map(async (d) => {
						if (isPnUser(d.wireJid)) {
						    // In the future we can make something to get multiple values from signal store at once (batch or something like that)
							const lid = await signalRepository.lidMapping.getLIDForPN(d.wireJid)
							if (lid) {
								logger.debug({ oldJid: d.wireJid, newJid: lid }, 'mapping PN to LID for session check')
								// IMPORTANT: We are MUTATING the wireJid here to its LID equivalent
								// This ensures all subsequent logic (assertSessions, createParticipantNodes)
								// uses the correct canonical JID.
								d.wireJid = lid
							}
						}
					})
				)
```

Reply:
Thanks for this smart idea, I'll change and test too